client/lib/i18n-utils/test/localize-path.js

```diff
@@ -0,0 +1,77 @@
+import { localizePath } from '../localize-path';
+
+describe( '#localizePath', () => {
+	test( 'should not change URL for `en`', () => {
+		[ '/', '/de/', '/start/', '/wp-login.php?action=lostpassword' ].forEach( ( fullUrl ) => {
+			expect( localizePath( fullUrl, 'en' ) ).toEqual( fullUrl );
+		} );
+	} );
+
+	test( 'should change relative URLs', () => {
+		expect( localizePath( '/fr/themes' ) ).toEqual( '/fr/themes/' );
+		expect( localizePath( '/fr/themes', 'en', false ) ).toEqual( '/fr/themes/' );
+		expect( localizePath( '/fr/themes', 'es', true ) ).toEqual( '/fr/themes/' );
+		expect( localizePath( '/fr/themes', 'fr', false ) ).toEqual( '/fr/themes/' );
+		expect( localizePath( '/fr/themes', 'es', false ) ).toEqual( '/fr/themes/' );
+		expect( localizePath( '/some/path', 'es', true ) ).toEqual( '/some/path/' );
+		expect( localizePath( '/some/path', 'es', false ) ).toEqual( '/es/some/path/' );
```

Comment:
Neither `localizePath` or `localizeUrl` should be localizing routes by default. Should only be localizing specific routes that support it. (/start, /themes, /plugins, /log-in etc)

```suggestion
		expect( localizePath( '/some/path', 'es', false ) ).toEqual( '/some/path/' );
```

Reply:
I'm not sure this should be the intended behaviour. There are a few unspecified URLs  which we still need to be localized:
* [https://wordpress.com/partners/](https://github.com/Automattic/wp-calypso/blob/trunk/client/layout/logged-out.jsx#L81)
* [https://wordpress.com/about/](https://github.com/Automattic/wp-calypso/blob/trunk/client/login/wp-login/index.jsx#L166)
* [https://wordpress.com/webinars](https://github.com/Automattic/wp-calypso/blob/trunk/client/me/help/main.jsx#L201)
* [https://wordpress.com/payments-donations/](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/earn/home.tsx#L274)
* [https://wordpress.com/learn/](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/contextual-help/contextual-help.tsx#L1570)

Although we can always include them to the `urlLocalizationMapping`, moving forward I think this will become hard to maintain. 
For example, there are plans to integrate the LOHP header and footer to Calypso logged-out pages (pau2Xa-4j7-p2). In this case, we'll need to specify and maintain all links to wordpress.com LPs.

Reply:
> I'm not sure this should be the intended behaviour.

I don't really have a strong opinion about this behavior, mostly just basing it off what `localize_url` and `localizeUrl` was setup to do (be fast and not make assumptions).

If you think you have enough context in calypso to make these assumptions I think you have an argument to add in something separate from `localizeUrl` in `localizePath`.

For LOHP pages it probably makes more sense to add the mappings to localizeUrl and port them to the php `localize_url` function as well.

Reply:
Thinking about it more, the `localizePath` the assumption would be that it receives only calypso related paths. Therefore, it probably should only localize whitelisted paths.

>  think you have an argument to add in something separate from localizeUrl in localizePath.

I found a workaround by setting an empty string for the first item in the `lookup` ([here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-utils/src/localize-url.tsx#L194)) array when  `localizeUrl` is called by `localizePath`. This should avoid it matching the `wordpress.com` segment and force it to only localize the whitelisted paths.